pipelinewise/cli/pipelinewise.py

```diff
@@ -1621,6 +1658,171 @@ def encrypt_string(self):
+                    python_bin=self.target_python_bin,
+                    config=cons_target_config,
+                )
+
+                transform_params = TransformParams(
+                    bin=self.transform_field_bin,
+                    config=tap_transformation,
+                    python_bin=self.transform_field_python_bin,
+                    tap_id=tap_id,
+                    target_id=target_id,
+                )
+
+                self.run_tap_partialsync(
+                    tap=tap_params, target=target_params, transform=transform_params,
+                )
+
+        except pidfile.AlreadyRunningError as exc:
+            self.logger.error('Another instance of the tap is already running.')
+            raise SystemExit(1) from exc
+        # Delete temp file if there is any
+        except commands.RunCommandException as exc:
+            self.logger.exception(exc)
+            self.send_alert(message=f'Failed to sync tables in {tap_id} tap', exc=exc)
+            raise SystemExit(1) from exc
+        except PartialSyncNotSupportedTypeException as exc:
+            self.logger.error(exc)
+            raise SystemExit(1) from exc
+        except PreRunChecksException as exp:
+            raise SystemExit(1) from exp
+        except Exception as exc:
+            self.send_alert(message=f'Failed to sync tables in {tap_id} tap', exc=exc)
+            raise exc
+        finally:
+            if cons_target_config:
+                utils.silentremove(cons_target_config)
+
+    def _check_supporting_tap_and_target_for_partial_sync(self):
+        tap_type = self.tap['type']
+        tap_id = self.tap['id']
+        target_type = self.target['type']
+        target_id = self.target['id']
+
+        if ConnectorType(target_type) not in PARTIAL_SYNC_PAIRS.get(ConnectorType(tap_type), {}):
+            raise PartialSyncNotSupportedTypeException(
+                f'Error! {tap_id}({tap_type})-{target_id}({target_type}) pair is not supported for the partial sync!'
+            )
+
+    def _check_if_complete_tap_configuration(self, fastsync_bin, tap_type, target_type):
+        # Tap exists but configuration not completed
+        if not os.path.isfile(fastsync_bin):
+            self.logger.error(
+                'Table sync function is not implemented from %s datasources to %s type of targets',
+                tap_type,
+                target_type
+            )
+            raise SystemExit(1)
+
+    def _check_if_tap_is_enabled(self):
+        # Run only if tap enabled
+        if not self.tap.get('enabled', False):
+            self.logger.info('Tap %s is not enabled.', self.tap['name'])
+            raise PreRunChecksException()
+
+    def _check_if_state_exists(self):
+        state_file = self.tap['files']['state']
+        if os.path.exists(state_file):
+            return
+        self.logger.error('Could not find state file in "%s"!', state_file)
+        raise PreRunChecksException()
```

Comment:
Would it be a problem if the state file exists but it's empty or a certain stream has no bookmark?

Reply:
there are two cases:
A) there is end_value: state file won't be updated and file existence means it was run before. that would be enough for running partial sync even if file is empty or damaged.
B) no end_value: state file will be updated. I trusted on the current existing method for updating state file. if it fails or can not update empty or damaged file then it can be a bug which needs its own tests and fixing in a separate task.